chain-vote/src/gang/p256k1.rs

```diff
@@ -45,23 +57,46 @@ impl GroupElement {
     /// Serialized GroupElement::zero
     const BYTES_ZERO: [u8; Self::BYTES_LEN] = [0; Self::BYTES_LEN];
 
-    pub fn generator() -> Self {
-        GroupElement(Point::generator())
+    /// Point from hash
+    pub fn from_hash(buffer: &[u8]) -> Self {
+        let mut result = [0u8; 33];
+        let mut hash = Blake2b::new(33);
+        let mut i = 0u32;
+        loop {
+            hash.input(buffer);
+            hash.input(&i.to_be_bytes());
+            hash.result(&mut result);
+            hash.reset();
+            // arbitrary encoding of sign
+            let sign = if result[32] & 1 == 0 {
+                Sign(Positive)
+            } else {
+                Sign(Negative)
+            };
```

Comment:
doesn't seem particularly useful to get the sign from the binary data and at the same time pass the binary data unmodified; why not just assume positive / negative value ?

Reply:
Wouldn't that make the output not uniform?

Reply:
Agree with @Zeegomo , assuming positive or negative would reduce the number of points we can randomly choose from. The sign should also be randomly selected. 

And I am not sure I understand the first part of your comment. We get the sign from the last byte, and then we pass all bytes but the last (unmodified) to compute the point. What is not useful about this?

Reply:
ok one thing I missed is that it's using blake2b264, not blake2b256. which make sense to get the bit from that extra byte. however, whilst it doesn't have the re-use bit that I originally thought about, it still have no practical use. losing one bit of randomness on E(Fp) make no difference in the grand scheme, and it doesn't make a difference on uniformity as you don't introduce a bias (all points / {0} have a negative, you just take away uniformly the negative).

Reply:
We made that choice to go according to the algorithm presented in [Short signatures from the Weil pairing](https://www.iacr.org/archive/asiacrypt2001/22480516.pdf) (Algorithm MapToGroup). And indeed, by preselecting the sign you choose uniformly from the negative, but we designed the algorithm so that it chooses uniformly from the prime order group. I agree that losing one bit of randomness makes not practical difference, but I would keep it like that unless you see an important drawback on this scheme.